Services/AzureManagementAdapter/AzureManagementAdapter.cs

```diff
@@ -121,14 +144,52 @@ private HttpRequest PrepareRequest(string path, string token, MetricsRequestList
             return request;
         }
 
+        /// <summary>
+        /// https://docs.microsoft.com/en-us/rest/api/monitor/autoscalesettings/createorupdate
+        /// </summary>
+        private HttpRequest PrepareVmssAutoscaleSettingsRequest(string token, string vmCount)
+        {
+            var autoScaleSettingsName = "scalevmss";
+            var request = new HttpRequest();
+            request.AddHeader(HttpRequestHeader.Accept.ToString(), "application/json");
+            request.AddHeader(HttpRequestHeader.CacheControl.ToString(), "no-cache");
+            request.AddHeader(HttpRequestHeader.Authorization.ToString(), token);
+            request.SetUriFromString($"{this.config.AzureManagementAdapterApiUrl}/{this.GetVmssAutoScaleSettingsUrl(autoScaleSettingsName)}");
+            request.Options.EnsureSuccess = false;
+            request.Options.Timeout = this.config.AzureManagementAdapterApiTimeout;
+            if (!this.config.AzureManagementAdapterApiUrl.ToLowerInvariant().StartsWith("https:"))
```

Comment:
this check should be in the config class and be thrown as soon as the service starts